src/test/java/edu/harvard/iq/dataverse/api/DataversesIT.java

```diff
@@ -1045,4 +1045,32 @@ public void testAddDataverse() {
                 .statusCode(BAD_REQUEST.getStatusCode())
                 .body("message", equalTo("Invalid metadata block name: \"" + invalidMetadataBlockName + "\""));
     }
+
+    @Test
+    public void testListFacets() {
+        Response createUserResponse = UtilIT.createRandomUser();
+        String apiToken = UtilIT.getApiTokenFromResponse(createUserResponse);
+        Response createDataverseResponse = UtilIT.createRandomDataverse(apiToken);
```

Comment:
Suggestion: instead of using createRandomDataverse and relying on the facets in "Root" to never change, maybe look at the test around lines 975-990 above to see how to set actual facets just in case root ever changes.

Reply:
Should probably get the latest from dev while you're at it.

Reply:
Good point, thanks. I have added a test case for collections with custom facets (not inherited from root) and refactored the code in the UtilIT class a bit.